mantle/platform/qemu.go

```diff
@@ -887,6 +881,59 @@ func (builder *QemuBuilder) setupUefi(secureBoot bool) error {
 	return nil
 }
 
+func (builder *QemuBuilder) setupIso() error {
+	if builder.ConfigFile != "" {
+		if builder.configInjected {
+			panic("config already injected?")
+		}
+		if err := builder.ensureTempdir(); err != nil {
+			return err
+		}
+		// TODO change to use something like an unlinked tempfile (or O_TMPFILE)
+		// in the same filesystem as the source so that reflinks (if available)
+		// will work
+		isoEmbeddedPath := filepath.Join(builder.tempdir, "install.iso")
+		cpcmd := exec.Command("cp", "--reflink=auto", builder.iso.path, isoEmbeddedPath)
+		cpcmd.Stderr = os.Stderr
+		if err := cpcmd.Run(); err != nil {
+			return errors.Wrapf(err, "copying iso")
+		}
+		configf, err := os.Open(builder.ConfigFile)
+		if err != nil {
+			return err
+		}
+		instCmd := exec.Command("coreos-installer", "iso", "embed", isoEmbeddedPath)
+		instCmd.Stdin = configf
+		instCmd.Stderr = os.Stderr
+		if err := instCmd.Run(); err != nil {
+			return errors.Wrapf(err, "running coreos-installer iso embed")
+		}
+		builder.iso.path = isoEmbeddedPath
+		builder.configInjected = true
+	}
+
+	// Arches s390x and ppc64le don't support UEFI and use the cdrom option to boot the ISO.
+	// For all other arches we use ide-cd device with bootindex=2 here: the idea is
+	// that during an ISO install, the primary disk isn't bootable, so the bootloader
+	// will fall back to the ISO boot. On reboot when the system is installed, the
+	// primary disk is selected. This allows us to have "boot once" functionality on
+	// both UEFI and BIOS (`-boot once=d` OTOH doesn't work with OVMF).
+
+	// TBD: aarch64 does not support ide-cd but cdrom won't work either.Including it here as it doesn't error out at least.
```

Comment:
I think it'd be better to explicitly error out with a message denoting that this setup path won't work on the arch.

Reply:
This code is just moved, but I can do that too.